web/src/pages/loading.tsx

```diff
@@ -37,18 +42,14 @@ const LoadingPage = (): ReactElement => {
         encounteredMyNjLinkingError: true,
       });
     } else {
-      if (loginPageEnabled) {
-        router && router.push(ROUTES.login);
-      } else {
-        triggerSignIn();
-      }
+      triggerSignIn();
     }
   }, [router, dispatch, loginPageEnabled]);
 
   useMountEffectWhenDefined(() => {
     if (!updateQueue) return;
     const business = updateQueue.currentBusiness();
-    if (!onboardingCompleted(business)) {
+    if (business && !onboardingCompleted(business)) {
```

Comment:
this was prematurely pushing users to `/onboarding` with some false positives

Reply:
I'm somewhat surprised this could even work without throwing an error. It looks like `onboardingCompleted` is expecting `business` to be defined.

```typescript
export const onboardingCompleted = (business: Business): boolean => {
  return business.onboardingFormProgress === "COMPLETED";
};
```